pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -627,43 +624,49 @@ def _getPenaltyParam(self, iw, rw):
-                objadd, fobj, fmerit, penparm, step, primalinf, dualinf, maxvi, maxvirel, hs, locj, indj, jcol, scales, bl, bu, fx, fcon, gcon, gobj, ycon,
-                pi, rc, rg, x, cu, iu, ru, cw, iw, rw):
+    def _snstop(self, ktcond, mjrprtlvl, minimize, n, nncon, nnobj, ns, itn, nmajor, nminor, nswap, condzhz, iobj,
+                scaleobj, objadd, fobj, fmerit, penparm, step, primalinf, dualinf, maxvi, maxvirel, hs, locj, indj,
+                jcol, scales, bl, bu, fx, fcon, gcon, gobj, ycon, pi, rc, rg, x, cu, iu, ru, cw, iw, rw):
         # fmt: on
         """
         This routine is called every major iteration in SNOPT, after solving QP but before line search
         We use it to determine the correct major iteration counting, and save some parameters in the history file.
-        If 'snSTOP function handle' is set to a function handle, then the callback is performed at the end of this function.
+        If 'snSTOP function handle' is set to a function handle, then it is called at the end of this function.
 
-        returning with iabort != 0 will terminate SNOPT immediately
+        Returns
+        -------
+        iabort : int
+            The return code expected by SNOPT. A non-zero value will terminate SNOPT immediately.
         """
         iterDict = {
             "isMajor": True,
             "nMajor": nmajor,
             "nMinor": nminor,
+            "step": step,
+            "feasibility": primalinf,
+            "optimality": dualinf,
+            "merit": fmerit,
+            "condZHZ": condzhz,
+            "penalty": penparm[2],
         }
         for saveVar in self.getOption("Save major iteration variables"):
-            if saveVar == "merit":
-                iterDict[saveVar] = fmerit
-            elif saveVar == "feasibility":
-                iterDict[saveVar] = primalinf
-            elif saveVar == "optimality":
-                iterDict[saveVar] = dualinf
-            elif saveVar == "penalty":
-                penParam = self._getPenaltyParam(iw, rw)
-                iterDict[saveVar] = penParam
+            if saveVar == "penalty_vector":
+                iterDict[saveVar] = self._getPenaltyParam(iw, rw),
             elif saveVar == "Hessian":
                 H = self._getHessian(iw, rw)
                 iterDict[saveVar] = H
-            elif saveVar == "step":
-                iterDict[saveVar] = step
-            elif saveVar == "condZHZ":
-                iterDict[saveVar] = condzhz
             elif saveVar == "slack":
                 iterDict[saveVar] = x[n:]
             elif saveVar == "lambda":
                 iterDict[saveVar] = pi
+            elif saveVar == "nS":
+                iterDict[saveVar] = ns
+            elif saveVar == "BSwap":
+                iterDict[saveVar] = nswap
+            elif saveVar == "maxVi":
+                iterDict[saveVar] = maxvi
+            else:
+                warnings.warn(f"Received unknown SNOPT save variable {saveVar}")
```

Comment:
I think an error would be better here

Reply:
I thought about it, and decided on warning since I don't want to kill a whole optimization just because someone made a typo or had a duplicate variable. Can change if preferred.

Reply:
Thoughts on this @lamkina?

Reply:
I would rather this be an error with a more descriptive message.  If you are expecting to have variables to postprocess afer an optimization, you don't want to have to re-run the entire optimization (especially for high-fidelity cases) just to get that information.  Especially because most users initially won't understand this warning and might just ignore the warning.

An error with a message like `Received unkown SNOPT save variable {saveVar}.  Please see "Save major iteration variables" option in the pyOptSparse documentation under "SNOPT".`